controllers/upgrade_handlers.go

```diff
@@ -21,36 +21,101 @@ import (
 	"fmt"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// handleUpgrade orchestrate main upgrade steps
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	// log and update status to in-progress
+	r.Log.Info("Starting handleUpgrade")
+	utils.SetStatusCondition(&ibu.Status.Conditions,
+		utils.GetInProgressConditionType(lcav1alpha1.Stages.Upgrade),
+		utils.ConditionReasons.InProgress,
+		metav1.ConditionTrue,
+		"Upgrade in-progress",
+		ibu.Generation)
```

Comment:
This will override the old status everytime it's re-queued and perform uncessary steps which were previously completed. We should check the current IBU status to determine if move to the next step.

Reply:
Yes, the stage handlers are not one shot. It can get called repeatedly if waiting for long running background jobs. The initial in progress condition is already set by the main controller when stage transition happens. The handlers can update the reason/message along the way until it reaches the end (status false with either success or failure)

Reply:
Removed it!